### ISSUE TITLE
Dim up and down?

### ISSUE THREAD

**Issue by 2oh1** [bwssytems/ha-bridge#903]
First, thank you so much for your work on HA Bridge.  It's fantastic!  I'm using it on a Raspberry Pi.

I only see one option in HA Bridge for Dim, but my Logitech Harmony Companion remote has a dimmer rocker with buttons to dim up and down (+ -).

Is there a way to pass on a command in that Dim section in HA Bridge that tells all of my Hue bulbs to increase or decrease brightness?

Here's a for instance as an example:

My Harmony Companion remote has no option for adding Hue Scenes to the Companion remote Home Control buttons (I've read that the Elite remote can add scenes there, but the Companion can't).  ...So, I created a bulb in HA Bridge called "Sunset."  Sunset triggers a warm light Hue scene.  It works for turning on the Sunset scene with my Companion remote, but the rocker dim buttons don't work - presumably because they're trying to dim a bulb that doesn't actually exist.

Is there a way to have dimming only function for physical bulbs?  Or, is there a way to pass on info that dictates which bulbs the dim command from the Harmony should be applied to?

**Comment by bwssytems** [comment]
For the harmony usage this way, do an edit/copy on your device in the ha-bridge and then set the "On with First Dim" to true.

**Comment by 2oh1** [comment]
I don't see an option for "On with First Dim."  I see things like Inactive, No State (Do not update state for device), Off State Resets Bri...  where in the device page on HA Bridge do I find On With First Dim?

**Comment by bwssytems** [comment]
Ahh, try the release candidate 5.2.0RC22 in #863

**Comment by tuicemen** [comment]
I thought that option only sent a on if the device was off?
I have several X10 devices which allow a dim and brighten the command is different for brighten then dim.

**Comment by bwssytems** [comment]
@2oh1 Any update on this?

**Comment by 2oh1** [comment]
For now, I've given up.  My issue is that Harmony Companion is crippled by design (the 'Home Control' buttons don't work the same way they do on the Harmony Elite, for example.  They can only trigger Hue bulbs, not scenes).  It's very laggy and occasionally unpredictable due to how Harmony treats the Companion 'Home Control' buttons.

I solved the bulbs/scenes issue with HA Bridge, but the remote is still sometimes terribly laggy for those Home Control buttons, and it can be unpredictable.

What I want is really simple.  I want the Home Control buttons to trigger Hue Scenes and I want the rocker switch to control brightness for all lights.  I don't want the rocker switch to reset any settings or remember any settings...  I just want it to pass along a command to increase or decrease brightness...  but the Harmony Companion gets confused because it tries to remember which button is currently active for the rocker and what the settings were, so it seems to keep multiple settings for the rocker in memory, and it gets confused if bulbs get triggered elsewhere (such as in an app).

The way Logitech implemented Hue into Harmony is awful.  Instead of working with Hue Scenes, Harmony imports scenes to create its own...  so, you end up with two versions of each scene depending on which device you're using.  It's a mess.

I gave up on the Companion and switched to a different remote - problem solved.
